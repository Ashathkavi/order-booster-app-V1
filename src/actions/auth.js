import database, {firebase, googleAuthProvider} from '../firebase/firebase'
import moment from 'moment'

export const login = (uid, name, role, createdAt) => ({
    type:'LOGIN',    
    uid,
    name,
    role,
    createdAt

})

export const startLogin = () => {
    return () => {
        return firebase.auth().signInWithPopup(googleAuthProvider)
    }
}


export const logout = () => ({
    type:'LOGOUT'
})

export const startLogout = () => {
    return () => {
        return firebase.auth().signOut()
    }
}

export const checkUserAvailability =  (user) => {
    
    return (dispatch) => {
        let userAvailabiliity = false
        return database.ref(`users/${user.uid}`).once('value')
            .then((snap)=>{
                userAvailabiliity = !!snap.val()
                console.log(userAvailabiliity, 'userAvailabiliity')
                if(userAvailabiliity){
                    dispatch(login(user.uid, snap.val().name, snap.val().role, snap.val().createdAt))
                }else{
                    dispatch(login(user.uid, user.displayName, 'unknown', 0))
                }
                console.log(userAvailabiliity, "userAvailabiliity")                
            })
            .catch((error)=>console.log('User availbaility request failed :', error))
        
        
    }
    
}


export const setAsMemeber = (userId, name) => {
    console.log(userId, "running set as member") 
    const createdAt = moment().valueOf()
    return (dispatch) => {
        return database.ref(`users/${userId}`).set({
            name,
            role: 'member',
            createdAt:createdAt
        }).then(()=>dispatch(login(userId, name, 'member', createdAt)))
            .catch((error)=>console.log('User creating request failed :', error))
    }
    
}