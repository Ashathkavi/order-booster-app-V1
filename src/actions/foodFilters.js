

//SET_NAME_FILTER : food filter reducer
export const setNameFilter = (name='') => ({
    type:'SET_NAME_FILTER',
    name
})

//SET_CAT_TEXT_FILTER : food filter reducer
export const setCatTextFilter = (catText='') => ({
    type:'SET_CAT_TEXT_FILTER',
    catText
})

//SET_BOUNDRY_AMOUNT : food filter reducer
export const setBoundryAmount = (boundryAmount=2000) => ({
    type:'SET_BOUNDRY_AMOUNT',
    boundryAmount
})


//SET_SIZE : food filter reducer
export const setSize = (size = '') => ({
    type: 'SET_SIZE',
    size
})

