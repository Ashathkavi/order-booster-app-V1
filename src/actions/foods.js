import uuid from 'uuid'
import moment from 'moment'
import database from '../firebase/firebase'

 
//CREATING ACTION GENERATORS..........................................

//ADD_FOOD : food reducer
export const addFood = (food)=>({
    type:'ADD_FOOD',
    food
})

export const startAddFood = (foodData = {})=> {
    return (dispatch) => {
        const {
            createdAt = moment().valueOf(),
            name='', 
            category = "soups", 
            amount=0, 
            largeAvailability=false, 
            foodSize='regular', 
            description=''
        } = foodData
        const food = {
            name,
            category,
            amount,
            largeAvailability,
            foodSize,
            description,
            createdAt    
        }
        return database.ref('foods').push(food)
            .then((refFood) => {
                console.log('Data is Saved')
                database.ref(`foods/${refFood.key}`).once('value').then((snap)=>{
                    if(!snap.val()){
                        dispatch(addFood({
                            id:refFood.key,
                            ...food
                        }))
                    }
                    
                })
                // dispatch(addFood({
                //     id:refFood.key,
                //     ...food
                // }))
            })
            .catch((error)=>console.log('failed :', error))
    }
}

//REMOVE_FOOD : food reducer
export const removeFood = ({id} = {}) => ({
    type:'REMOVE_FOOD',
    id
})


export const startRemoveFood = ({id} = {})=> {
    return (dispatch) => {        
        return database.ref(`foods/${id}`).remove()
            .then(() => {
                console.log('Data is removed')
                dispatch(removeFood({id}))
            })
            .catch((error)=>console.log('failed :', error))
    }
}



//EDIT_FOOD : food reducer
export const editFood = (id, updates) => ({
    type:'EDIT_FOOD',
    id,
    updates

})

export const startEditFood = (id, updates)=> {
    return (dispatch) => {        
        return database.ref(`foods/${id}`).update(updates)
            .then(() => {
                console.log('Data is updated')
                dispatch(editFood(id, updates))
            })
            .catch((error)=>console.log('failed :', error))
    }
}



//SET_FOODS : to get all foods detail from database
export const setFoods = (foods) => ({
    type:'SET_FOODS',
    foods
})

export const startSetFoods = ()=> {
    return (dispatch) => {
        
        return database.ref('foods').once('value').then((snapshot) => {
            console.log('Data is fetched')
            const foods = []
            snapshot.forEach((childSnapshot)=>{
                foods.push({
                    
                    ...childSnapshot.val(),
                    id:childSnapshot.key
                })
            })
            dispatch(setFoods(foods))
        }).catch((e)=>{
            console.log('Error with data fetching: ', e)
        })

    }
}


export const onStartSetFoods = ()=> {
    return (dispatch) => {
        return database.ref('foods').on('value', (snapshot) => {
            const foods = []
            snapshot.forEach((childSnapshot)=>{
                foods.push({
                    ...childSnapshot.val(),
                    id:childSnapshot.key
                })
            })
            dispatch(setFoods(foods))
        })
    }
    
}


