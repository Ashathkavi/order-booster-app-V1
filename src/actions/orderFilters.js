

//SET_ADDRESS_FILTER : order filter reducer
export const setAddressFilter = (address='') => ({
    type:'SET_ADDRESS_FILTER',
    address
})

//SET_CUSTOMER_FILTER : order filter reducer
export const setCustomerFilter = (customerName='') => ({
    type:'SET_CUSTOMER_FILTER',
    customerName
})


//SET_BOUNDRY_AMOUNT : order filter reducer
export const setBoundryAmount = (boundryAmount=2000) => ({
    type:'SET_BOUNDRY_AMOUNT',
    boundryAmount
})

//SET_FOOD_FILTER : order filter reducer
export const setFoodFilter = (food='') => ({
    type:'SET_FOOD_FILTER',
    food
})

//SET_P_NO_FILTER : order filter reducer
export const setPNoFilter = (phoneNumber='') => ({
    type:'SET_P_NO_FILTER',
    phoneNumber
})

//SET_DELIVERMETH_FILTER : order filter reducer
export const setDeliverMethFilter = (deliverMeth='') => ({
    type:'SET_DELIVERMETH_FILTER',
    deliverMeth
})

//SET_STATUS_FILTER : order filter reducer
export const setStatusFilter = (status='') => ({
    type:'SET_STATUS_FILTER',
    status
})

// //SET_KOT_STATUS_FILTER : order filter reducer
// export const setKotStatusFilter = (kotStatus='') => ({
//     type:'SET_KOT_STATUS_FILTER',
//     kotStatus
// })

// //SET_BILL_STATUS_FILTER : order filter reducer
// export const setBillStatusFilter = (billStatus='') => ({
//     type:'SET_BILL_STATUS_FILTER',
//     billStatus
// })

//SORT_BY_DATE : order filter reducer
export const sortByDate = () => ({
    type:'SORT_BY_DATE'
}) 
//SORT_BY_AMOUNT : order filter reducer
export const sortByAmount = () => ({
    type:'SORT_BY_AMOUNT'
}) 
//SORT_BY_DURATION : order filter reducer
export const sortByDuration = () => ({
    type:'SORT_BY_DURATION'
})
//SET_START_DATE : order filter reducer
export const setStartDate = (startDate = undefined) => ({
    type:'SET_START_DATE',
    startDate
}) 
//SET_END_DATE : order filter reducer
export const setEndDate = (endDate = undefined) => ({
    type:'SET_END_DATE',
    endDate
}) 
