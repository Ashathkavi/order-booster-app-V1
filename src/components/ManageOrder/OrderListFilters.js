import React, {useState, useEffect} from 'react'
import {connect} from 'react-redux'
import moment from 'moment'
import {
    setBoundryAmount,
    setFoodFilter,
    // setBillStatusFilter,
    // setKotStatusFilter,
    setStatusFilter,
    setPNoFilter,
    setCustomerFilter,
    setAddressFilter,
    setStartDate,
    setEndDate,
    sortByDuration,
    sortByAmount,
    sortByDate,
    setDeliverMethFilter
} from '../../actions/orderFilters' 
import DatePicker from 'react-datepicker'



export const OrderListFilters = (props) => {

    const [boundryAmountDisplay, setboundryAmountDisplay] = useState('')
    const [displayManageFilter, setDisplayManageFilter] = useState(false)

    useEffect(()=>{
        if(props.autherizedAs === 'admin' || props.autherizedAs === 'manager'){
            setDisplayManageFilter(true)
        }
    },[])


    const onDateChange = (dates) => {
        let [startDate, endDate] = dates;
        startDate = moment(startDate).valueOf()
        endDate =  moment(endDate).valueOf()
        //console.log(startDate,'start',endDate,'end')
        props.setStartDate(startDate)
        props.setEndDate(endDate)
    }

    const onAddressChange = (e) => {
        props.setAddressFilter(e.target.value)
    }

    const onCustomerChange = (e) => {
        props.setCustomerFilter(e.target.value)
    }

    const onFoodChange = (e) => {
        props.setFoodFilter(e.target.value)
    }

    const onPhoneNumberChange = (e) => {
        props.setPNoFilter(e.target.value)
    }

    const onBoundryAmountChange = (e) => {
        //console.log(typeof e.target.value )
        const boundryAmount = parseInt(e.target.value)
        //console.log(boundryAmount )
        props.setBoundryAmount(boundryAmount)

        setboundryAmountDisplay(e.target.value)
    }

    const onStatusChange = (e) => {
        props.setStatusFilter(e.target.value)
    }

    const onDeliverMethChange = (e) => {
        
        props.setDeliverMethFilter(e.target.value)
    }

    // const onKotStatusChange = (e) => {
    //     props.setKotStatusFilter(e.target.value)
    // }

    // const onBillStatusChange = (e) => {
    //     props.setBillStatusFilter(e.target.value)
    // }

    const onSortChange = (e) => {
        const billStatus = e.target.value
                
        if (billStatus=== 'duration'){
            props.sortByDuration()
        }else if(billStatus === 'amount'){
            props.sortByAmount()
        }else if(billStatus === 'date'){
            props.sortByDate()
        } 
    }

    


    return displayManageFilter ? (
        <div className = "content-container" >
            <div className="input-group">
                <button onClick={()=>props.setSawFilters(false)}>Hide Filter</button>
            </div>
            <div className="input-group">
                <div className="input-group__item">
                    {/*setAddressFilter filter*/}
                    <input className="text-input" type="text" value={props.filters.address} placeholder="Type an Address" onChange={onAddressChange}/>
                </div>
                <div className="input-group__item">
                    {/*setCustomerFilter filter*/}
                    <input className="text-input" type="text" value={props.filters.customerName} placeholder="Type a Customer Name" onChange={onCustomerChange}/>
                </div>
                <div className="input-group__item" >
                    {/*setFoodFilter filter*/}
                    <input className="text-input" type="text" value={props.filters.food} placeholder="Type a Food" onChange={onFoodChange}/>
                </div>
                <div className="input-group__item">
                    {/*setPNoFilter filter*/}
                    <input className="text-input" type="number" value={props.filters.phoneNumber} pattern={/^[0-9]{10}$/} placeholder="Type a PhoneNumber" onChange={onPhoneNumberChange}/>
                </div>
                
            </div>
            <div className="input-group">
                <div className="input-group__item input-group__item--range">
                    {/*setBoundryAmount filter*/}                    
                    <div>                    
                        <input className="text-input" type='text' id='textInput' disabled value={boundryAmountDisplay}></input>
                    </div>
                    <div>                    
                        <input className="text-input" type='range' min='0' max='20000' step='100' value={props.filters.boundryAmount} onChange={onBoundryAmountChange}/>
                    </div>
                </div>
                <div className="input-group__item">
                    {/*setStatusFilter filter*/}
                    <select className="select" value={props.filters.status} onChange={onStatusChange}>
                        <option value=''>All</option>
                        <option value="confirmed">Confirmed</option>
                        <option value="kitchen">Kitchen</option>
                        <option value="table">Table</option>
                        <option value="cancelled">Cancelled</option>
                        <option value="on delivery">On Delivery</option>
                        <option value="recieved">Recieved</option>
                    </select>
                </div>
                <div className="input-group__item"> 
                    {/*setBillStatusFilter filter*/}
                    <select className="select" value={props.filters.sortBy} onChange={onSortChange} placeholder="abcd">            
                        <option value="date">Date</option>
                        <option value="duration">Available Time</option>
                        <option value="amount">Amount</option>
                    </select>
                </div>
                <div className="input-group__item"> 
                    {/*setBillStatusFilter filter*/}
                    <select className="select" value={props.filters.deliverMeth} onChange={onDeliverMethChange} >
                        <option value="">Any</option>            
                        <option value="delivery">Delivery</option>
                        <option value="take away">Take Away</option>
                        <option value="dinning">Dinning</option>
                    </select>
                </div>
                <div className="input-group__item">
                    {/*setStartDate setEndDate filter*/}
                    <DatePicker    
                        selected={props.filters.startDate}
                        startDate={props.filters.startDate}   
                        endDate={props.filters.endDate} 
                        selectsRange
                        onChange= {onDateChange}
                        className="datePicker"
                    />  
                </div>
            </div>
        </div>
)  : (
        <div className = "content-container" >
            <div className="input-group">
                <button onClick={()=>props.setSawFilters(false)}>Hide Filter</button>
            </div>
            <div className="input-group">
                <div className="input-group__item" >
                    {/*setFoodFilter filter*/}
                    <input className="text-input" type="text" value={props.filters.food} placeholder="Type a Food" onChange={onFoodChange}/>
                </div>
                <div className="input-group__item">
                    {/*setStatusFilter filter*/}
                    <select className="select" value={props.filters.status} onChange={onStatusChange}>
                        <option value=''>All</option>
                        <option value="confirmed">Confirmed</option>
                        <option value="kitchen">Kitchen</option>
                        <option value="table">Table</option>
                        <option value="cancelled">Cancelled</option>
                        <option value="on delivery">On Delivery</option>
                        <option value="recieved">Recieved</option>
                    </select>
                </div>
                <div className="input-group__item"> 
                    {/*setBillStatusFilter filter*/}
                    <select className="select" value={props.filters.sortBy} onChange={onSortChange} placeholder="abcd">            
                        <option value="date">Date</option>
                        <option value="duration">Available Time</option>
                        <option value="amount">Amount</option>
                    </select>
                </div>
                <div className="input-group__item"> 
                    {/*setBillStatusFilter filter*/}
                    <select className="select" value={props.filters.deliverMeth} onChange={onDeliverMethChange} >
                        <option value="">Any</option>            
                        <option value="delivery">Delivery</option>
                        <option value="take away">Take Away</option>
                        <option value="dinning">Dinning</option>
                    </select>
                </div>
                <div className="input-group__item">
                    {/*setStartDate setEndDate filter*/}
                    <DatePicker    
                        selected={props.filters.startDate}
                        startDate={props.filters.startDate}   
                        endDate={props.filters.endDate} 
                        selectsRange
                        onChange= {onDateChange}
                        className="datePicker"
                    />  
                </div>
            </div>
        </div>

)}

const mapStateToProps = (state) =>  ({
    filters:state.orderFilters,
    autherizedAs:state.auth.role
})

const mapDispatchToProps = (dispatch) => ({
    setAddressFilter: (address) => dispatch(setAddressFilter(address)),
    setCustomerFilter: (customerName) => dispatch(setCustomerFilter(customerName)),
    setBoundryAmount: (amount) => dispatch(setBoundryAmount(amount)),
    setFoodFilter: (food) => dispatch(setFoodFilter(food)),
    setPNoFilter: (phoneNumber) => dispatch(setPNoFilter(phoneNumber)),
    setDeliverMethFilter:(deliverMeth) => dispatch(setDeliverMethFilter(deliverMeth)),
    setStatusFilter: (status) => dispatch(setStatusFilter(status)),
    setKotStatusFilter: (kotStatus) => dispatch(setKotStatusFilter(kotStatus)),
    setBillStatusFilter: (billStatus) => dispatch(setBillStatusFilter(billStatus)),    
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    sortByDuration: () => dispatch(sortByDuration()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),

})


export default connect(mapStateToProps, mapDispatchToProps)(OrderListFilters)