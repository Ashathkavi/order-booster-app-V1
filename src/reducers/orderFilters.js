import moment from 'moment'

//ORDERFILTER REDUCER
const orderFilterReducerDefaultState = {
    address:'',
    customerName:'',
    boundryAmount:20000,
    food:'',
    phoneNumber:'',
    status:'',
    kotStatus:'',
    billStatus:'',
    sortBy:'date', // date or amount or duration
    startDate:moment().startOf('month').valueOf(),
    endDate:moment().endOf('month').valueOf()

}
const orderFilterReducer = (state=orderFilterReducerDefaultState, action) => {
    switch(action.type){
        case 'SET_ADDRESS_FILTER':
            return {...state,address:action.address}
        case 'SET_CUSTOMER_FILTER':
            return {...state,customerName:action.customerName}
        case 'SET_BOUNDRY_AMOUNT':
            return {...state,boundryAmount:action.boundryAmount}
        case 'SET_FOOD_FILTER':
            return {...state,food:action.food}
        case 'SET_P_NO_FILTER':
            return {...state,phoneNumber:action.phoneNumber}
        case 'SET_STATUS_FILTER':
            return {...state,status:action.status}
        case 'SET_KOT_STATUS_FILTER':
            return {...state,kotStatus:action.kotStatus}
        case 'SET_BILL_STATUS_FILTER':
            return {...state,billStatus:action.billStatus }
        case 'SORT_BY_DATE':
            return {...state, sortBy:'date'}
        case 'SORT_BY_AMOUNT':
            return {...state, sortBy:'amount'}
        case 'SORT_BY_DURATION':
            return {...state, sortBy:'duration'}
        case 'SET_START_DATE':
            return {...state, startDate:action.startDate}
        case 'SET_END_DATE':
            return {...state, endDate:action.endDate}
    default:
            return state
    }
}

export default orderFilterReducer