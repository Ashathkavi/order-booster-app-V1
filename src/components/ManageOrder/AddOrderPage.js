import React from 'react'
import {OrderForm} from './OrderForm'
import {startAddOrder} from '../../actions/orders'
import {connect} from 'react-redux'
import selectFood from '../../selectors/foods'



export const AddOrderPage = (props) => {

    
    const counter = (orders) =>{
        let count = orders.length + 1
        let i = true
        while(i === true){

            orders.map((order) => {
                i=false
                if(order.count  === count){
                    count += 1
                    i=true
                }
            })
            if(orders.length === 0){
                i = false
            }
        }
        return count
    } 

    const onSubmit = (order) => {
        props.startAddOrder({...order, count:counter(props.orders)})
        props.history.push('/order')
    }

    return(
        <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Add Order</h1>
                </div>
            </div>

            <div className="content-container">
                <OrderForm orders={props.orders} onSubmit={ onSubmit} foods={props.foods}/>
            </div>
            

        </div>
)}


const mapDispatchToProps = (dispatch) => ({
    startAddOrder: (order) => dispatch(startAddOrder(order))
})


const mapStateToProps = (state) => {
    console.log(state.orders)
    return {
        orders:state.orders,
        foods:selectFood(state.foods, state.foodFilters)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddOrderPage)