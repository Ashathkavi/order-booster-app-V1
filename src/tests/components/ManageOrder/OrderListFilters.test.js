import React from 'react'
import {shallow} from 'enzyme'
import {OrderListFilters} from '../../../components/ManageOrder/OrderListFilters'
import {orderFilter, altOrderFilter} from '../../../fixtures/sampleOrderFilters'
import DatePicker from 'react-datepicker'
import moment from 'moment'
import { setStartDate } from '../../../actions/orderFilters'

let setAddressFilterSpy, setCustomerFilterSpy, setBoundryAmountSpy, setFoodFilterSpy
let wrapper, setPNoFilterSpy, setStatusFilterSpy, setKotStatusFilterSpy, setBillStatusFilterSpy
let sortByDateSpy, sortByAmountSpy, sortByDurationSpy, setStartDateSpy, setEndDateSpy


beforeEach(() => {
    setAddressFilterSpy = jest.fn()
    setCustomerFilterSpy = jest.fn()
    setBoundryAmountSpy = jest.fn()
    setFoodFilterSpy = jest.fn()
    setPNoFilterSpy = jest.fn()
    setStatusFilterSpy = jest.fn()
    setKotStatusFilterSpy = jest.fn()
    setBillStatusFilterSpy = jest.fn()
    sortByDateSpy = jest.fn()
    sortByAmountSpy = jest.fn()
    sortByDurationSpy = jest.fn()
    setStartDateSpy = jest.fn()
    setEndDateSpy = jest.fn()
    wrapper = shallow(
        <OrderListFilters 
            filters={orderFilter}
            setAddressFilter = {setAddressFilterSpy}
            setCustomerFilter = {setCustomerFilterSpy}
            setBoundryAmount = {setBoundryAmountSpy}
            setFoodFilter = {setFoodFilterSpy}
            setPNoFilter = {setPNoFilterSpy}
            setStatusFilter = {setStatusFilterSpy}
            setKotStatusFilter = {setKotStatusFilterSpy}
            setBillStatusFilter = {setBillStatusFilterSpy}
            sortByDate = {sortByDateSpy}
            sortByAmount = {sortByAmountSpy}
            sortByDuration = {sortByDurationSpy}
            setStartDate = {setStartDateSpy}
            setEndDate = {setEndDateSpy}
        />
    )
})

test('should render OrderList Coorectly', ()=>{
    expect(wrapper).toMatchSnapshot()
})

test('should render OrderList with alternate order filter', ()=>{
    wrapper.setProps({
        filters:altOrderFilter
    })
    expect(wrapper).toMatchSnapshot()
})


test('should handle setAddressFilter', () => {
    const value = 'mama'
    wrapper.find('input').at(0).simulate('change', {
        
        persist: () => { },
        target:{value}
    })
    expect(setAddressFilterSpy).toHaveBeenLastCalledWith(value)
})


test('should handle setCustomerFilter', () => {
    const value = 'push'
    wrapper.find('input').at(1).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setCustomerFilterSpy).toHaveBeenLastCalledWith(value)
})


test('should handle setFoodFilterS', () => {
    const value = 'chi'
    wrapper.find('input').at(2).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setFoodFilterSpy).toHaveBeenLastCalledWith(value)
})

test('should handle setPNoFilter', () => {
    const value = 'chi'
    wrapper.find('input').at(3).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setPNoFilterSpy).toHaveBeenLastCalledWith(value)
})

test('should handle setboundryAmount', () => {
    const value = '0756'
    wrapper.find('input').at(4).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setBoundryAmountSpy).toHaveBeenLastCalledWith(value)
})


test('should handle setStatusFilter', () => {
    const value = 'conf'
    wrapper.find('select').at(0).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setStatusFilterSpy).toHaveBeenLastCalledWith(value)
})


test('should handle setKotStatusFilter', () => {
    const value = 'conf'
    wrapper.find('select').at(1).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setKotStatusFilterSpy).toHaveBeenLastCalledWith(value)
})


test('should handle setBillStatusFilter', () => {
    const value = 'conf'
    wrapper.find('select').at(2).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(setBillStatusFilterSpy).toHaveBeenLastCalledWith(value)
})


test('should handle sortbyamount', () => {
    const value = 'amount'
    wrapper.find('select').at(3).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(sortByAmountSpy).toHaveBeenCalled()
})


test('should handle sortByDate', () => {
    const value = 'date'
    wrapper.find('select').at(3).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(sortByDateSpy).toHaveBeenCalled()
})


test('should handle sortbyduration', () => {
    const value = 'duration'
    wrapper.find('select').at(3).simulate('change', {
        persist: () => { },
        target:{value}
    })
    expect(sortByDurationSpy).toHaveBeenCalled()
})





test('should handle category modal clossing', () => {
    const startDate = moment().startOf('month')
    const endDate =moment().endOf('month')
    wrapper.find(DatePicker).prop('onChange')([startDate.toDate(), endDate.toDate()]);    
    expect(setStartDateSpy).toHaveBeenLastCalledWith(startDate.valueOf())
    expect(setEndDateSpy).toHaveBeenLastCalledWith(endDate.valueOf())
})

