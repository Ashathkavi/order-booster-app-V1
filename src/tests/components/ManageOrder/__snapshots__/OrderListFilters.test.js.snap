// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OrderList Coorectly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Type an Address"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Type a Customer Name"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Type a Food"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    pattern={/\\^\\[0-9\\]\\{10\\}\\$/}
    placeholder="Type a PhoneNumber"
    type="number"
    value=""
  />
  <input
    max="20000"
    min="0"
    onChange={[Function]}
    step="100"
    type="range"
    value={20000}
  />
  <input
    disabled={true}
    id="textInput"
    type="text"
    value=""
  />
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="confirmed"
    >
      Confirmed
    </option>
    <option
      value="kitchen"
    >
      Kitchen
    </option>
    <option
      value="table"
    >
      Table
    </option>
    <option
      value="cancelled"
    >
      Cancelled
    </option>
    <option
      value="on delivery"
    >
      On Delivery
    </option>
    <option
      value="recieved"
    >
      Recieved
    </option>
  </select>
  <select
    onChange={[Function]}
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="passed"
    >
      Passed
    </option>
    <option
      value="not"
    >
      Not Passed
    </option>
  </select>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="printed"
    >
      Printed
    </option>
    <option
      value="not"
    >
      Not Printed
    </option>
  </select>
  <select
    onChange={[Function]}
    value="date"
  >
    <option
      value="date"
    >
      Date
    </option>
    <option
      value="duration"
    >
      Available Time
    </option>
    <option
      value="amount"
    >
      Amount
    </option>
  </select>
  <r
    allowSameDay={false}
    calendarStartDay={0}
    customTimeInput={null}
    dateFormat="MM/dd/yyyy"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    enableTabLoop={true}
    endDate={2658599999}
    excludeScrollbar={true}
    focusSelectedMonth={false}
    monthsShown={1}
    nextMonthButtonLabel="Next Month"
    nextYearButtonLabel="Next Year"
    onBlur={[Function]}
    onCalendarClose={[Function]}
    onCalendarOpen={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    previousYearButtonLabel="Previous Year"
    readOnly={false}
    renderDayContents={[Function]}
    selected={-19800000}
    selectsRange={true}
    shouldCloseOnSelect={true}
    showFourColumnMonthYearPicker={false}
    showFullMonthYearPicker={false}
    showMonthYearPicker={false}
    showPopperArrow={true}
    showPreviousMonths={false}
    showQuarterYearPicker={false}
    showTimeInput={false}
    showTimeSelect={false}
    showTwoColumnMonthYearPicker={false}
    showYearPicker={false}
    startDate={-19800000}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    withPortal={false}
    yearItemNumber={12}
  />
</div>
`;

exports[`should render OrderList with alternate order filter 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Type an Address"
    type="text"
    value="mama"
  />
  <input
    onChange={[Function]}
    placeholder="Type a Customer Name"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Type a Food"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    pattern={/\\^\\[0-9\\]\\{10\\}\\$/}
    placeholder="Type a PhoneNumber"
    type="number"
    value=""
  />
  <input
    max="20000"
    min="0"
    onChange={[Function]}
    step="100"
    type="range"
    value={20000}
  />
  <input
    disabled={true}
    id="textInput"
    type="text"
    value=""
  />
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="confirmed"
    >
      Confirmed
    </option>
    <option
      value="kitchen"
    >
      Kitchen
    </option>
    <option
      value="table"
    >
      Table
    </option>
    <option
      value="cancelled"
    >
      Cancelled
    </option>
    <option
      value="on delivery"
    >
      On Delivery
    </option>
    <option
      value="recieved"
    >
      Recieved
    </option>
  </select>
  <select
    onChange={[Function]}
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="passed"
    >
      Passed
    </option>
    <option
      value="not"
    >
      Not Passed
    </option>
  </select>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="printed"
    >
      Printed
    </option>
    <option
      value="not"
    >
      Not Printed
    </option>
  </select>
  <select
    onChange={[Function]}
    value="date"
  >
    <option
      value="date"
    >
      Date
    </option>
    <option
      value="duration"
    >
      Available Time
    </option>
    <option
      value="amount"
    >
      Amount
    </option>
  </select>
  <r
    allowSameDay={false}
    calendarStartDay={0}
    customTimeInput={null}
    dateFormat="MM/dd/yyyy"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    enableTabLoop={true}
    endDate={2658599999}
    excludeScrollbar={true}
    focusSelectedMonth={false}
    monthsShown={1}
    nextMonthButtonLabel="Next Month"
    nextYearButtonLabel="Next Year"
    onBlur={[Function]}
    onCalendarClose={[Function]}
    onCalendarOpen={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    previousYearButtonLabel="Previous Year"
    readOnly={false}
    renderDayContents={[Function]}
    selected={-19800000}
    selectsRange={true}
    shouldCloseOnSelect={true}
    showFourColumnMonthYearPicker={false}
    showFullMonthYearPicker={false}
    showMonthYearPicker={false}
    showPopperArrow={true}
    showPreviousMonths={false}
    showQuarterYearPicker={false}
    showTimeInput={false}
    showTimeSelect={false}
    showTwoColumnMonthYearPicker={false}
    showYearPicker={false}
    startDate={-19800000}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    withPortal={false}
    yearItemNumber={12}
  />
</div>
`;
